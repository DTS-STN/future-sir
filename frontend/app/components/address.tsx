import type { ComponentProps, JSX } from 'react';

import { formatAddress } from '~/utils/string-utils';
import { cn } from '~/utils/tailwind-utils';

type Address = {
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  postalZipCode?: string;
  provinceState?: string;
  country: string;
};

type AddressProps = ComponentProps<'address'> & {
  address: Address;
  /**
   * The format of the address
   *
   * - `standard`: The standard address format, with the address line, city, province/state, postal/zip code, and country.
   * - `alternative`: An alternative address format, with the address line, city, province/state, postal/zip code, and country on separate lines.
   */
  format?: 'standard' | 'alternative';
};

export function Address({ address, format, className, ...rest }: AddressProps): JSX.Element {
  return (
    <address className={cn('whitespace-pre-wrap not-italic', className)} data-testid="address-id" {...rest}>
      {formatAddress(address, format)}
    </address>
  );
}
