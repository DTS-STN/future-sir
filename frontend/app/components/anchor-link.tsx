import type { ComponentProps, JSX, MouseEvent } from 'react';

/**
 * AnchorLinkProps represents the properties for the AnchorLink component.
 * It extends the ComponentProps<'a'> type, omitting the 'href' property,
 * and adds the required 'anchorElementId' property.
 */
export interface AnchorLinkProps extends OmitStrict<ComponentProps<'a'>, 'href'> {
  anchorElementId: string;
}

/**
 * AnchorLink is a React component used to create anchor links that scroll
 * and focus on the specified target element when clicked.
 *
 * @param props - The properties for the AnchorLink component.
 * @returns React element representing the anchor link.
 */
export function AnchorLink({ anchorElementId, children, onClick, ...restProps }: AnchorLinkProps): JSX.Element {
  /**
   * handleOnSkipLinkClick is the click event handler for the anchor link.
   * It prevents the default anchor link behavior, scrolls to and focuses
   * on the target element specified by 'anchorElementId', and invokes
   * the optional 'onClick' callback.
   */
  function handleOnSkipLinkClick(event: MouseEvent<HTMLAnchorElement>) {
    event.preventDefault();
    scrollAndFocusFromAnchorLink(event.currentTarget.href);
    onClick?.(event);
  }

  return (
    <a href={`#${anchorElementId}`} onClick={handleOnSkipLinkClick} data-testid="anchor-link" {...restProps}>
      {children}
    </a>
  );
}

/**
 * Scrolls and focuses on the element identified by the anchor link's hash.
 *
 * @param href - The anchor link URL.
 */
function scrollAndFocusFromAnchorLink(href: string): void {
  if (URL.canParse(href)) {
    const { hash } = new URL(href);

    if (hash) {
      const targetElement = document.querySelector<HTMLElement>(hash);

      if (targetElement) {
        targetElement.scrollIntoView({ behavior: 'smooth' });
        targetElement.focus();
      }
    }
  }
}
