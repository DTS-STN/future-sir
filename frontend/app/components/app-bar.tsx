import { useTranslation } from 'react-i18next';

import { LanguageSwitcher } from '~/components/language-switcher';
import { Menu } from '~/components/menu';
import { UserButton } from '~/components/user-button';

interface AppBarProps {
  children: React.ReactNode;
  name?: string;
  profileItems?: React.ReactNode;
}

export function AppBar({ children, name, profileItems }: AppBarProps) {
  const { t } = useTranslation(['gcweb']);

  return (
    <div className="bg-slate-700">
      <div className="align-center container mx-auto flex flex-wrap justify-between">
        <div className="align-center flex">
          <Menu>{children}</Menu>
        </div>
        <div className="flex items-center space-x-4 text-right">
          {name && <UserButton name={name}>{profileItems}</UserButton>}
          <LanguageSwitcher>{t('gcweb:language-switcher.alt-lang')}</LanguageSwitcher>
        </div>
      </div>
    </div>
  );
}
