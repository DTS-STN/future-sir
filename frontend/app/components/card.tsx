import type { ComponentProps, HTMLElementType, ReactElement } from 'react';

import type { To } from 'react-router';

import type { FlipProp, IconProp } from '@fortawesome/fontawesome-svg-core';
import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { AppLink } from '~/components/links';
import type { I18nRouteFile } from '~/i18n-routes';
import { cn } from '~/utils/tailwind-utils';

type CardBaseProps = Omit<ComponentProps<typeof AppLink>, 'to' | 'file' | 'title'> & {
  to?: To;
  file?: I18nRouteFile;
};

type CardProps = CardBaseProps & {
  /**
   * Optional image to be displayed at the top of the card.
   */
  image?: ReactElement<typeof CardImage>;

  /**
   * Optional tag to be displayed at the top-left corner of the card.
   */
  tag?: ReactElement<typeof CardTag>;
};

/**
 * Card component renders a card with an optional image, icon, tag.
 * It supports various customization options and can be used for navigation or clicking
 */
export function Card({ children, className, image, tag, ...props }: CardProps) {
  const { disabled, to, file, onClick } = props;
  const isLink = to || file ? true : false;
  const isButton = onClick ? true : false;
  const isClickable = isLink || isButton;
  const baseClassName = `rounded-sm ${isClickable ? 'select-none' : ''} ${disabled ? 'bg-white' : 'bg-blue-900'}`;

  return (
    <div className={cn(baseClassName, className)}>
      <CardBase isLink={isLink} isButton={isButton} {...props}>
        {image}
        <div className="relative p-8">
          {tag}
          <div className="flex items-center gap-4">{children}</div>
        </div>
      </CardBase>
    </div>
  );
}

/**
 * Provides the base element for the Card component
 * Renders an AppLink if the 'To' or 'File' prop is specified.
 * Renders an Anchor tag if the 'onClick' prop is provided.
 * Defaults to rendering a Span tag if neither prop is given.
 */
function CardBase({
  disabled,
  isLink,
  isButton,
  ...props
}: CardBaseProps & {
  isLink?: boolean;
  isButton?: boolean;
}) {
  const baseClassName = cn(
    'group block overflow-hidden rounded-sm border border-gray-500 bg-white shadow-xs outline-offset-4',
    'aria-disabled:pointer-events-none aria-disabled:bg-gray-200 aria-disabled:opacity-70 aria-disabled:shadow-none',
  );
  const interactableClassName =
    'transition-opacity duration-200 ease-in-out hover:bg-gray-50 hover:shadow-md focus:bg-gray-50 active:opacity-90';

  return isLink ? (
    <AppLink
      className={cn(baseClassName, interactableClassName)}
      disabled={disabled}
      {...(props as ComponentProps<typeof AppLink>)}
    >
      {props.children}
    </AppLink>
  ) : isButton ? (
    <a role="button" className={cn(baseClassName, interactableClassName)} aria-disabled={disabled} tabIndex={0} {...props}>
      {props.children}
    </a>
  ) : (
    <span className={baseClassName} aria-disabled={disabled} {...props}>
      {props.children}
    </span>
  );
}

type CardImageProps = {
  /**
   * Optional image URL to be displayed at the top of the card.
   */
  src: string;

  /**
   * Alternative text for the image.
   */
  alt: string;
};

export function CardImage({ src, alt }: CardImageProps) {
  return (
    <img
      src={src}
      alt={alt}
      className="relative aspect-video h-40 w-full object-cover sm:h-48"
      loading="lazy"
      draggable="false"
    />
  );
}

type CardTagProps = {
  /**
   * Optional tag text to be displayed at the top-left corner of the card.
   */
  tag: string;
};

export function CardTag({ tag }: CardTagProps) {
  return <span className="absolute top-0 left-0 block bg-cyan-700 px-2 py-1 text-xs text-white">{tag}</span>;
}

type CardIconProps = {
  /**
   * Optional FontAwesome icon to be displayed.
   */
  icon: IconProp;

  /**
   * The flip property for the icon.
   */
  iconFlip?: FlipProp;

  /**
   * Children property, used for the card content
   */
  children?: React.ReactNode;
};

export function CardIcon({ icon, iconFlip, children }: CardIconProps) {
  return (
    <div className="flex items-center gap-4">
      <div className="flex size-12 shrink-0 items-center justify-center rounded-full bg-slate-700">
        <FontAwesomeIcon icon={icon} flip={iconFlip} className="size-5 text-white" />
      </div>
      {children}
    </div>
  );
}

type CardTitleProps = {
  /**
   * The title text to be displayed.
   */
  title: string;

  /**
   *  The HTML element to be used for the title
   *  @default 'h3'
   */
  titleAs?: ExtractStrict<HTMLElementType, 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p'>;

  /**
   * Determines if it highlights like a link.
   */
  highlight?: boolean;

  /**
   * Children property, used for the card content
   */
  children?: React.ReactNode;
};

export function CardTitle({ titleAs: TitleComponent = 'h3', title, highlight, children }: CardTitleProps) {
  const baseClassName = `text-lg font-bold text-slate-700 ${highlight ? 'underline group-hover:text-blue-700 group-focus:text-blue-700' : ''}`;
  return (
    <div className="space-y-2">
      <div className="flex items-center gap-2">
        <TitleComponent className={baseClassName}>{title}</TitleComponent>
        {highlight && <FontAwesomeIcon aria-hidden icon={faChevronRight} className={cn('mt-1', baseClassName)} />}
      </div>
      <div>{children}</div>
    </div>
  );
}
