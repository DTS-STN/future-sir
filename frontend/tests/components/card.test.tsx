import { createRoutesStub } from 'react-router';

import { faUserPlus } from '@fortawesome/free-solid-svg-icons';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Card, CardIcon, CardImage, CardTag, CardTitle } from '~/components/card';

describe('Card', () => {
  it('should render a card', () => {
    const { container } = render(
      <Card>
        <CardIcon icon={faUserPlus}>
          <CardTitle title="Card Title" highlight>
            Test Card
          </CardTitle>
        </CardIcon>
      </Card>,
    );
    expect(container.innerHTML).toMatchSnapshot('expected html');
  });

  it('should render a disabled card', () => {
    const { container } = render(
      <Card disabled>
        <CardIcon icon={faUserPlus}>
          <CardTitle title="Card Title" highlight>
            Test Card
          </CardTitle>
        </CardIcon>
      </Card>,
    );
    expect(container.innerHTML).toMatchSnapshot('expected html');
  });

  it('should render a card with a link', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/fr/public',
        Component: () => (
          <Card file="routes/public/index.tsx">
            <CardIcon icon={faUserPlus}>
              <CardTitle title="Card Title" highlight>
                Test Card
              </CardTitle>
            </CardIcon>
          </Card>
        ),
      },
    ]);

    const { container } = render(<RoutesStub initialEntries={['/fr/public']} />);
    expect(container.innerHTML).toMatchSnapshot('expected html');
  });

  it('should render a card with a link with an image', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/fr/public',
        Component: () => (
          <Card
            file="routes/public/index.tsx"
            image={<CardImage src="https://www.canada.ca/content/dam/canada/activities/20250115-1-520x200.jpg" alt="" />}
            tag={<CardTag tag={'Coming soon'} />}
          >
            <CardIcon icon={faUserPlus}>
              <CardTitle title="Card Title" highlight>
                Test Card
              </CardTitle>
            </CardIcon>
          </Card>
        ),
      },
    ]);

    const { container } = render(<RoutesStub initialEntries={['/fr/public']} />);
    expect(container.innerHTML).toMatchSnapshot('expected html');
  });

  it('should render a disabled card with a link', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/fr/public',
        Component: () => (
          <Card
            disabled
            file="routes/public/index.tsx"
            image={<CardImage src="https://www.canada.ca/content/dam/canada/activities/20250115-1-520x200.jpg" alt="" />}
            tag={<CardTag tag={'Coming soon'} />}
          >
            <CardIcon icon={faUserPlus}>
              <CardTitle title="Card Title" highlight>
                Test Card
              </CardTitle>
            </CardIcon>
          </Card>
        ),
      },
    ]);

    const { container } = render(<RoutesStub initialEntries={['/fr/public']} />);
    expect(container.innerHTML).toMatchSnapshot('expected html');
  });

  it('should render a card with onClick', () => {
    const RoutesStub = createRoutesStub([
      {
        Component: () => (
          <Card onClick={() => {}} tag={<CardTag tag={'Coming soon'} />}>
            <CardIcon icon={faUserPlus}>
              <CardTitle title="Card Title" highlight>
                Test Card
              </CardTitle>
            </CardIcon>
          </Card>
        ),
      },
    ]);

    const { container } = render(<RoutesStub initialEntries={['/fr/public']} />);
    expect(container.innerHTML).toMatchSnapshot('expected html');
  });

  it('should render a card with onClick with an image', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/fr/public',
        Component: () => (
          <Card
            onClick={() => {}}
            image={<CardImage src="https://www.canada.ca/content/dam/canada/activities/20250115-1-520x200.jpg" alt="" />}
            tag={<CardTag tag={'Coming soon'} />}
          >
            <CardIcon icon={faUserPlus}>
              <CardTitle title="Card Title" highlight>
                Test Card
              </CardTitle>
            </CardIcon>
          </Card>
        ),
      },
    ]);

    const { container } = render(<RoutesStub initialEntries={['/fr/public']} />);
    expect(container.innerHTML).toMatchSnapshot('expected html');
  });

  it('should render a disabled card with onClick', () => {
    const RoutesStub = createRoutesStub([
      {
        path: '/fr/public',
        Component: () => (
          <Card
            disabled
            onClick={() => {}}
            image={<CardImage src="https://www.canada.ca/content/dam/canada/activities/20250115-1-520x200.jpg" alt="" />}
            tag={<CardTag tag={'Coming soon'} />}
          >
            <CardIcon icon={faUserPlus}>
              <CardTitle title="Card Title" highlight>
                Test Card
              </CardTitle>
            </CardIcon>
          </Card>
        ),
      },
    ]);

    const { container } = render(<RoutesStub initialEntries={['/fr/public']} />);
    expect(container.innerHTML).toMatchSnapshot('expected html');
  });
});
