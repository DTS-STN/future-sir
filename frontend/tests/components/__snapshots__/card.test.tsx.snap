// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > should render a card > expected html 1`] = `"<div class="rounded-sm bg-blue-900"><span class="group block overflow-hidden rounded-sm border border-gray-500 bg-white shadow-xs outline-offset-4 aria-disabled:pointer-events-none aria-disabled:bg-gray-200 aria-disabled:opacity-70 aria-disabled:shadow-none"><div class="relative p-8"><div class="flex items-center gap-4"><div class="flex items-center gap-4"><div class="flex size-12 shrink-0 items-center justify-center rounded-full bg-slate-700"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="user-plus" class="svg-inline--fa fa-user-plus size-5 text-white" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><path fill="currentColor" d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304l91.4 0C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM504 312l0-64-64 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l64 0 0-64c0-13.3 10.7-24 24-24s24 10.7 24 24l0 64 64 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-64 0 0 64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"></path></svg></div><div class="space-y-2"><div class="flex items-center gap-2"><h3 class="text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700">Card Title</h3><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="chevron-right" class="svg-inline--fa fa-chevron-right mt-1 text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path fill="currentColor" d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"></path></svg></div><div>Test Card</div></div></div></div></div></span></div>"`;

exports[`Card > should render a card with a link > expected html 1`] = `"<div class="rounded-sm select-none bg-blue-900"><a class="group block overflow-hidden rounded-sm border border-gray-500 bg-white shadow-xs outline-offset-4 aria-disabled:pointer-events-none aria-disabled:bg-gray-200 aria-disabled:opacity-70 aria-disabled:shadow-none transition-opacity duration-200 ease-in-out hover:bg-gray-50 hover:shadow-md focus:bg-gray-50 active:opacity-90" href="/fr/public" data-discover="true"><div class="relative p-8"><div class="flex items-center gap-4"><div class="flex items-center gap-4"><div class="flex size-12 shrink-0 items-center justify-center rounded-full bg-slate-700"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="user-plus" class="svg-inline--fa fa-user-plus size-5 text-white" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><path fill="currentColor" d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304l91.4 0C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM504 312l0-64-64 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l64 0 0-64c0-13.3 10.7-24 24-24s24 10.7 24 24l0 64 64 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-64 0 0 64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"></path></svg></div><div class="space-y-2"><div class="flex items-center gap-2"><h3 class="text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700">Card Title</h3><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="chevron-right" class="svg-inline--fa fa-chevron-right mt-1 text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path fill="currentColor" d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"></path></svg></div><div>Test Card</div></div></div></div></div></a></div>"`;

exports[`Card > should render a card with a link with an image > expected html 1`] = `"<div class="rounded-sm select-none bg-blue-900"><a class="group block overflow-hidden rounded-sm border border-gray-500 bg-white shadow-xs outline-offset-4 aria-disabled:pointer-events-none aria-disabled:bg-gray-200 aria-disabled:opacity-70 aria-disabled:shadow-none transition-opacity duration-200 ease-in-out hover:bg-gray-50 hover:shadow-md focus:bg-gray-50 active:opacity-90" href="/fr/public" data-discover="true"><img alt="" class="relative aspect-video h-40 w-full object-cover sm:h-48" loading="lazy" draggable="false" src="https://www.canada.ca/content/dam/canada/activities/20250115-1-520x200.jpg"><div class="relative p-8"><span class="absolute top-0 left-0 block bg-cyan-700 px-2 py-1 text-xs text-white">Coming soon</span><div class="flex items-center gap-4"><div class="flex items-center gap-4"><div class="flex size-12 shrink-0 items-center justify-center rounded-full bg-slate-700"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="user-plus" class="svg-inline--fa fa-user-plus size-5 text-white" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><path fill="currentColor" d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304l91.4 0C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM504 312l0-64-64 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l64 0 0-64c0-13.3 10.7-24 24-24s24 10.7 24 24l0 64 64 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-64 0 0 64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"></path></svg></div><div class="space-y-2"><div class="flex items-center gap-2"><h3 class="text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700">Card Title</h3><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="chevron-right" class="svg-inline--fa fa-chevron-right mt-1 text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path fill="currentColor" d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"></path></svg></div><div>Test Card</div></div></div></div></div></a></div>"`;

exports[`Card > should render a card with onClick > expected html 1`] = `"<h2>Unexpected Application Error!</h2><h3 style="font-style: italic;">404 Not Found</h3><p>ðŸ’¿ Hey developer ðŸ‘‹</p><p>You can provide a way better UX than this when your app throws errors by providing your own <code style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);">ErrorBoundary</code> or <code style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);">errorElement</code> prop on your route.</p>"`;

exports[`Card > should render a card with onClick with an image > expected html 1`] = `"<div class="rounded-sm select-none bg-blue-900"><a role="button" class="group block overflow-hidden rounded-sm border border-gray-500 bg-white shadow-xs outline-offset-4 aria-disabled:pointer-events-none aria-disabled:bg-gray-200 aria-disabled:opacity-70 aria-disabled:shadow-none transition-opacity duration-200 ease-in-out hover:bg-gray-50 hover:shadow-md focus:bg-gray-50 active:opacity-90" tabindex="0"><img alt="" class="relative aspect-video h-40 w-full object-cover sm:h-48" loading="lazy" draggable="false" src="https://www.canada.ca/content/dam/canada/activities/20250115-1-520x200.jpg"><div class="relative p-8"><span class="absolute top-0 left-0 block bg-cyan-700 px-2 py-1 text-xs text-white">Coming soon</span><div class="flex items-center gap-4"><div class="flex items-center gap-4"><div class="flex size-12 shrink-0 items-center justify-center rounded-full bg-slate-700"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="user-plus" class="svg-inline--fa fa-user-plus size-5 text-white" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><path fill="currentColor" d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304l91.4 0C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM504 312l0-64-64 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l64 0 0-64c0-13.3 10.7-24 24-24s24 10.7 24 24l0 64 64 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-64 0 0 64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"></path></svg></div><div class="space-y-2"><div class="flex items-center gap-2"><h3 class="text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700">Card Title</h3><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="chevron-right" class="svg-inline--fa fa-chevron-right mt-1 text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path fill="currentColor" d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"></path></svg></div><div>Test Card</div></div></div></div></div></a></div>"`;

exports[`Card > should render a disabled card > expected html 1`] = `"<div class="rounded-sm bg-white"><span class="group block overflow-hidden rounded-sm border border-gray-500 bg-white shadow-xs outline-offset-4 aria-disabled:pointer-events-none aria-disabled:bg-gray-200 aria-disabled:opacity-70 aria-disabled:shadow-none" aria-disabled="true"><div class="relative p-8"><div class="flex items-center gap-4"><div class="flex items-center gap-4"><div class="flex size-12 shrink-0 items-center justify-center rounded-full bg-slate-700"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="user-plus" class="svg-inline--fa fa-user-plus size-5 text-white" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><path fill="currentColor" d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304l91.4 0C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM504 312l0-64-64 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l64 0 0-64c0-13.3 10.7-24 24-24s24 10.7 24 24l0 64 64 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-64 0 0 64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"></path></svg></div><div class="space-y-2"><div class="flex items-center gap-2"><h3 class="text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700">Card Title</h3><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="chevron-right" class="svg-inline--fa fa-chevron-right mt-1 text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path fill="currentColor" d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"></path></svg></div><div>Test Card</div></div></div></div></div></span></div>"`;

exports[`Card > should render a disabled card with a link > expected html 1`] = `"<div class="rounded-sm select-none bg-white"><a role="link" aria-disabled="true" class="group block overflow-hidden rounded-sm border border-gray-500 bg-white shadow-xs outline-offset-4 aria-disabled:pointer-events-none aria-disabled:bg-gray-200 aria-disabled:opacity-70 aria-disabled:shadow-none transition-opacity duration-200 ease-in-out hover:bg-gray-50 hover:shadow-md focus:bg-gray-50 active:opacity-90"><img alt="" class="relative aspect-video h-40 w-full object-cover sm:h-48" loading="lazy" draggable="false" src="https://www.canada.ca/content/dam/canada/activities/20250115-1-520x200.jpg"><div class="relative p-8"><span class="absolute top-0 left-0 block bg-cyan-700 px-2 py-1 text-xs text-white">Coming soon</span><div class="flex items-center gap-4"><div class="flex items-center gap-4"><div class="flex size-12 shrink-0 items-center justify-center rounded-full bg-slate-700"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="user-plus" class="svg-inline--fa fa-user-plus size-5 text-white" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><path fill="currentColor" d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304l91.4 0C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM504 312l0-64-64 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l64 0 0-64c0-13.3 10.7-24 24-24s24 10.7 24 24l0 64 64 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-64 0 0 64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"></path></svg></div><div class="space-y-2"><div class="flex items-center gap-2"><h3 class="text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700">Card Title</h3><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="chevron-right" class="svg-inline--fa fa-chevron-right mt-1 text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path fill="currentColor" d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"></path></svg></div><div>Test Card</div></div></div></div></div></a></div>"`;

exports[`Card > should render a disabled card with onClick > expected html 1`] = `"<div class="rounded-sm select-none bg-white"><a role="button" class="group block overflow-hidden rounded-sm border border-gray-500 bg-white shadow-xs outline-offset-4 aria-disabled:pointer-events-none aria-disabled:bg-gray-200 aria-disabled:opacity-70 aria-disabled:shadow-none transition-opacity duration-200 ease-in-out hover:bg-gray-50 hover:shadow-md focus:bg-gray-50 active:opacity-90" aria-disabled="true" tabindex="0"><img alt="" class="relative aspect-video h-40 w-full object-cover sm:h-48" loading="lazy" draggable="false" src="https://www.canada.ca/content/dam/canada/activities/20250115-1-520x200.jpg"><div class="relative p-8"><span class="absolute top-0 left-0 block bg-cyan-700 px-2 py-1 text-xs text-white">Coming soon</span><div class="flex items-center gap-4"><div class="flex items-center gap-4"><div class="flex size-12 shrink-0 items-center justify-center rounded-full bg-slate-700"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="user-plus" class="svg-inline--fa fa-user-plus size-5 text-white" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><path fill="currentColor" d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304l91.4 0C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM504 312l0-64-64 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l64 0 0-64c0-13.3 10.7-24 24-24s24 10.7 24 24l0 64 64 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-64 0 0 64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"></path></svg></div><div class="space-y-2"><div class="flex items-center gap-2"><h3 class="text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700">Card Title</h3><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="chevron-right" class="svg-inline--fa fa-chevron-right mt-1 text-lg font-bold text-slate-700 underline group-hover:text-blue-700 group-focus:text-blue-700" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path fill="currentColor" d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"></path></svg></div><div>Test Card</div></div></div></div></div></a></div>"`;
